mgc/discriminability/oneSample.py

```diff
@@ -0,0 +1,107 @@
+        H_A: D_X > D_0 
+    
+    where D_0 is the discriminability that would be observed by random chance.
+    
+    References
+    ----------
+     .. [#1Dscr] Eric W. Bridgeford, et al. "Optimal Decisions for Reference Pipelines and Datasets: Applications in Connectomics." Bioarxiv (2019).
+    """
+
+    def __init__(self):
+        self.stat = None
+        self.pvalue = None
+        DiscriminabilityTest.__init__(self)
+
+
+
+    def test(self, X, Y, isDist = False, remove_isolates=True, reps=1000, workers=-1):
+        r"""
+        Calculates the Discriminability one sample test statistic and p-value.
+
+        Parameters
+        ----------
+        x, y : ndarray
+            Input data matrices. `x` and `y` must have the same number of
+            samples. That is, the shapes must be `(n, p)` and `(n, q)` where
+            `n` is the number of samples and `p` and `q` are the number of
+            dimensions. Alternatively, `x` and `y` can be distance matrices,
+            where the shapes must both be `(n, n)`.
+        isDist : Boolean, optional (default: False)
+                 Whether `x` is a distance matrix.
+        remove_isolates : Boolean, optional (default: True)
+                          whether remove the samples with single instance.
+        reps : int, optional (default: 1000)
+            The number of replications used to estimate the null distribution
+            when using the permutation test used to calculate the p-value.
+        workers : int, optional (default: -1)
+            The number of cores to parallelize the p-value computation over.
+            Supply -1 to use all cores available to the Process.
+
+        Returns
+        -------
+        stat : float
+            The computed Discriminability statistic.
+        pvalue : float
+            The computed one sample test p-value.
+        """
+
+        check_X_y(X, Y, accept_sparse=True)
+
+        _, counts = np.unique(Y, return_counts=True)
+
+        if (counts != 1).sum() <= 1:
+            msg = "You have passed a vector containing only a single unique sample id."
+            raise ValueError(msg)
+
+
+        self.X = X
+        self.Y = Y
+        
+        stat = super(oneSample,self)._statistic(self.X, self.Y,isDist, remove_isolates, return_rdfs=False)
+        self.stat = stat
+
+        # use all cores to create function that parallelizes over number of reps
+        mapwrapper = MapWrapper(workers)
+        null_dist = np.array(list(mapwrapper(self._perm_stat, range(reps))))
+        self.null_dist = null_dist
+
+        # calculate p-value and significant permutation map through list
+        pvalue = (null_dist >= stat).sum() / reps
+
+        # correct for a p-value of 0. This is because, with bootstrapping
+        # permutations, a p-value of 0 is incorrect
+        if pvalue == 0:
+            pvalue = 1 / reps
+        self.pvalue = pvalue
+
+        return stat, pvalue
+
+
+
+
+    def _perm_stat(self, index):
```

Comment:
This is a basic permutation common to most functions in `mgc`, so move to abstract class as mentioned before